packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp

```diff
@@ -358,27 +396,69 @@ namespace MueLu {
     LO num_bad_aggs = 0;
     MT worst_agg = 0.0;
 
+    MT mean_bad_agg = 0.0;
+    MT mean_good_agg  = 0.0;
+
+
     for (size_t i=0;i<agg_qualities->getLocalLength();++i) {
 
-      if (data[i] > good_agg_thresh) num_bad_aggs++;
+      if (data[i] > good_agg_thresh) {
+        num_bad_aggs++;
+        mean_bad_agg += data[i];
+      }
+      else {
+        mean_good_agg += data[i];
+      }
       worst_agg = std::max(worst_agg, data[i]);
-
     }
 
+
+    if (num_bad_aggs > 0) mean_bad_agg /= num_bad_aggs;
+    mean_good_agg /= agg_qualities->getLocalLength() - num_bad_aggs;
+
     if (num_bad_aggs == 0) {
-      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << std::endl;
+      GetOStream(Statistics1) << "All aggregates passed the quality measure. Worst aggregate had quality " << worst_agg << ". Mean aggregate quality " << mean_good_agg << "." << std::endl;
     } else {
-      GetOStream(Statistics1) << num_bad_aggs << " out of " << agg_qualities->getLocalLength() << " did not pass the quality measure. Worst aggregate had quality " << worst_agg << std::endl;
+      GetOStream(Statistics1) << num_bad_aggs << " out of " << agg_qualities->getLocalLength() << " did not pass the quality measure. Worst aggregate had quality " << worst_agg << ". "
+                              << "Mean bad aggregate quality " << mean_bad_agg << ". Mean good aggregate quality " << mean_good_agg << "." << std::endl;
```

Comment:
`mean_good_agg` and `mean_bad_agg` are local quantities.   If an MPI process other than root has `num_bad_aggs>0`, it won't print will it, because GetOStream(Statistics1) only prints from the root process?

Reply:
Correct.